lib/matplotlib/tests/test_backend_bases.py

```diff
@@ -286,3 +287,59 @@ def test_draw(backend):
 
     for ref, test in zip(layed_out_pos_agg, layed_out_pos_test):
         np.testing.assert_allclose(ref, test, atol=0.005)
+
+
+@pytest.mark.parametrize(
+    "key,mouseend,expectedxlim,expectedylim",
+    [(None, (0.2, 0.2), (3.49, 12.49), (2.7, 11.7)),
+     (None, (0.2, 0.5), (3.49, 12.49), (0, 9)),
+     (None, (0.5, 0.2), (0, 9), (2.7, 11.7)),
+     (None, (0.5, 0.5), (0, 9), (0, 9)),  # No move
+     (None, (0.8, 0.25), (-3.47, 5.53), (2.25, 11.25)),
+     (None, (0.2, 0.25), (3.49, 12.49), (2.25, 11.25)),
+     (None, (0.8, 0.85), (-3.47, 5.53), (-3.14, 5.86)),
+     (None, (0.2, 0.85), (3.49, 12.49), (-3.14, 5.86)),
+     ("shift", (0.2, 0.4), (3.49, 12.49), (0, 9)),  # snap to x
+     ("shift", (0.4, 0.2), (0, 9), (2.7, 11.7)),  # snap to y
+     ("shift", (0.2, 0.25), (3.49, 12.49), (3.49, 12.49)),  # snap to diagonal
+     ("shift", (0.8, 0.25), (-3.47, 5.53), (3.47, 12.47)),  # snap to diagonal
+     ("shift", (0.8, 0.9), (-3.58, 5.41), (-3.58, 5.41)),  # snap to diagonal
+     ("shift", (0.2, 0.85), (3.49, 12.49), (-3.49, 5.51)),  # snap to diagonal
+     ("x", (0.2, 0.1), (3.49, 12.49), (0, 9)),  # only x
+     ("y", (0.1, 0.2), (0, 9), (2.7, 11.7)),  # only y
+     ("control", (0.2, 0.2), (3.49, 12.49), (3.49, 12.49)),  # diagonal
+     ("control", (0.4, 0.2), (2.72, 11.72), (2.72, 11.72)),  # diagonal
+     ])
+def test_interactive_pan(key, mouseend, expectedxlim, expectedylim):
+    fig, ax = plt.subplots()
+    ax.plot(np.arange(10))
+    assert ax.get_navigate()
+    # Set equal aspect ratio to easier see diagonal snap
+    ax.set_aspect('equal')
+
+    # Mouse move starts from 0.5, 0.5
+    mousestart = (0.5, 0.5)
+    # Convert to screen coordinates ("s").  Events are defined only with pixel
+    # precision, so round the pixel values, and below, check against the
+    # corresponding xdata/ydata, which are close but not equal to d0/d1.
+    sstart = ax.transData.transform(mousestart).astype(int)
+    send = ax.transData.transform(mouseend).astype(int)
+
+    # Set up the mouse movements
+    start_event = MouseEvent(
+        "button_press_event", fig.canvas, *sstart, button=MouseButton.LEFT,
+        key=key)
+    stop_event = MouseEvent(
+        "button_release_event", fig.canvas, *send, button=MouseButton.LEFT,
+        key=key)
+
+    tb = NavigationToolbar2(fig.canvas)
+    tb.pan()
+    tb.press_pan(start_event)
+    tb.drag_pan(stop_event)
+    tb.release_pan(stop_event)
+    print(ax.get_xlim())
```

Comment:
I should get rid of these... (Although they only print on errors...)

Reply:
does that get incorporated into the error message printed by the assert?

Reply:
Sort of. If the test fails, anything on stdout will be shown under a "Captured stdout" or something similar heading. (I used this to get the actual values for the test...).